src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -426,75 +394,36 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 	if (workerJob->requiresMasterEvaluation)
 	{
 		PlanState *planState = &(scanState->customScanState.ss.ps);
+		EState *executorState = planState->state;
 
 		ExecuteMasterEvaluableFunctions(jobQuery, planState);
 
+		/* we've already processed parameters */
+		executorState->es_param_list_info = NULL;
```

Comment:
Yes, this was to avoid redundantly sending the values even though they're already in the deserialised query.